### PR TITLE
Exclude compact.xbf from package if it belongs to other project

### LINKED ISSUE #599: Payload contains two or more files with the same destination path Compact.xbf
**Describe the bug**
UWP and UWP Xunit test project in same solution, both referencing Microsoft.UI.Xaml >= 2.1.190405004, causes error "Payload contains two or more files with the same destination path 'Microsoft.UI.Xaml\DensityStyles\Compact.xbf" (in each project's \bin\x86\Debug\ folder).

If I downgrade both project's Microsoft.UI.Xaml to 2.0.181018004 the error goes away.

**Steps to reproduce the bug**
I was trying Windows Template Studio for VS 2019 so these might differ slightly:

1. Add a Universal Windows C# project to your solution.
2. Add a Universal Windows C# xUnit test project to your solution.
3. Check NuGet Package Manager and make sure Microsoft.UI.Xaml is using version 2.1.x on both projects.
4. Build the solution.

**Expected behavior**
The project should build with no errors.

**Version Info**

NuGet package version: 
Microsoft.UI.Xaml 2.1.190405004.1

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Fall Creators Update (16299)     | yes   |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | yes   |

**Additional context**
Visual Studio 2019 16.0.2
Solution generated with Windows Template Studio.
Target Windows version: 1809 (17763)
Min Windows version: Fall Creators Update (16299)

### PR DESCRIPTION
**Pull request by licanhua**
Fix #599
To support compact, I injected a to the project for framework package. It introduced problem in _ComputeAppxPackagePayload if more than two projects are references the microsoft.ui.xaml and the two projects have reference relationship too.
For example, we have two projects: C# xUnit project and C# project:
C# xUnit project reference MUX nuget, it copied Compact.xaml to obj and compiled it.
C# project reference MUX nuget, it copied Compact.xaml again.
C# xUnit project reference C# project.
When compile C# xUnit project, _ComputeAppxPackagePayload aggregate Compact.xaml from both project in target _ComputeAppxPackagePayload, and task RemoveDuplicates detects this conflicts.

This just exclude the compact.xbf from packaging if it's matching with the MSBuildProjectName

### CODE REVIEW

File: build/NuSpecs/MUXControls-Nuget-FrameworkPackage.props [hunk 1]
```diff
@@ -113,6 +114,15 @@
     </ItemGroup>
   </Target>
 
+  <!-- Workaround for https://github.com/Microsoft/microsoft-ui-xaml/issues/599. Exclude Compact.xbf which is not in this project -->
+  <Target Name="_FixXamlCompactXbfPackaging" BeforeTargets="_ComputeAppxPackagePayload">
```
Comment (jevansaks) [inline build/NuSpecs/MUXControls-Nuget-FrameworkPackage.props]: Can you elaborate on what this is doing? What is the problem and how is this condition fixing it?

Comment (jevansaks) [reply build/NuSpecs/MUXControls-Nuget-FrameworkPackage.props]: The description has some more details but this sentence confused me: "This just exclude the compact.xbf from packaging if it's matching with the MSBuildProjectName". I think this should have said if the app being packaged *doesn't match* MSBuildProjectName? 

---
In reply to: [278214089](https://github.com/Microsoft/microsoft-ui-xaml/pull/610#discussion_r278214089) [](ancestors = 278214089)

File: build/NuSpecs/MUXControls-Nuget-FrameworkPackage.props [hunk 2]
```diff
@@ -113,6 +114,15 @@
     </ItemGroup>
   </Target>
 
+  <!-- Workaround for https://github.com/Microsoft/microsoft-ui-xaml/issues/599. Exclude Compact.xbf which is not in this project -->
```
Comment (licanhua) [inline build/NuSpecs/MUXControls-Nuget-FrameworkPackage.props]:
```suggestion
  <!-- Workaround for https://github.com/Microsoft/microsoft-ui-xaml/issues/599. 
  If there are more than two projects are using microsoft-ui-xaml framework package, and the projects have reference relationship,
  more than two compact.xbf with the same destination path would be packaged and compiler would complains error. 
  To make sure only one compact.xbf is packaged, this target excludes the compact.xbf which doesn't match MSBuildProjectName.
  --> 
```

**Comment by jevansaks** [comment]
Please also add a test to the test\MUXControlsReleaseTest project somewhere to verify that the compact dictionary is appearing and the documented URL works as expected in a consuming app. The project gets built and run against both the dll-based and framework pakage-based nuget packages so this would exercise both configurations and make sure this continues to work going forward. Thanks!

**Pull request merged**